api/v1alpha1/database_webhook.go

```diff
@@ -188,10 +193,35 @@ func instanceSpecValidatorForCreate(instance *Instance, allErrs field.ErrorList,
 		))
 	}
 
+	if isInvalidAdditionalArguments, allowedAdditionalArguments := isAdditionalArgumentsInvalid(instance.Type, instance.AdditionalArguments); isInvalidAdditionalArguments {
+		allErrs = append(allErrs, field.Invalid(instancePath.Child("additionalArguments"), instance.AdditionalArguments,
+			fmt.Sprintf("Additional Arguments for %s are invalid! Valid values are: %s", instance.Type, reflect.ValueOf(allowedAdditionalArguments).MapKeys()),
+		))
+	}
+
 	databaselog.Info("Exiting instanceSpecValidatorForCreate...")
 	return allErrs
 }
 
+/* Checks if configured additional arguments are invalid */
+func isAdditionalArgumentsInvalid(typ string, additionalArguments map[string]string) (bool, map[string]bool) {
```

Comment:
I would suggest renaming 'typ' to dbType

Reply:
Sounds good!